a2lfile/src/a2ml.rs

```diff
@@ -337,8 +409,9 @@ fn tokenize_a2ml(input: &str) -> Result<Vec<TokenType>, String> {
 // parse an a2ml fragment in an a2l file
 // The target data structure is the parsing definition used by the a2l parser, so that the
 // a2ml can control the parsing of IF_DATA blocks
-pub(crate) fn parse_a2ml(input: &str) -> Result<A2mlTypeSpec, String> {
-    let tok_result = tokenize_a2ml(input)?;
+pub(crate) fn parse_a2ml(filename: Option<String>, input: &str) -> Result<(A2mlTypeSpec, String), String> {
+    let mut complete_string = "".to_string();
```

Comment:
This should be

    let mut complete_string = String::with_capacity(input.len());

That way it will not reallocate while it grows in the common case where the a2ml does not contain any /includes

Reply:
done